introducing-pgzx.mdx

```diff
@@ -0,0 +1,208 @@
+---
+title: 'Introducing pgzx: create PostgreSQL extensions using Zig'
+description: 'We are excited to introduce pgzx, a framework for creating Postgres extensions using the Zig programming language.'
+image:
+  src: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/introducing-pgroll.jpg
+  alt: pgzx
+author: Tudor Golubenco, Steffen Siering
+authorEmail: tudor@xata.io
+date: 3-21-2024
+tags: ['open-source', 'postgres', 'fpPgzx']
+published: true
+slug: introducing-pgzx
+ogImage: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/introducing-pgroll.jpg
+---
+
+As part of the [Xata launch week](https://xata.io/launch-week-unleash-the-elephant), we are introducing [pgzx](https://github.com/xataio/pgzx), an open-source framework for developing PostgreSQL extensions written in Zig. It provides a set of utilities (e.g. error handling, memory allocators, wrappers) as well as a build and development environment that simplify integrating with the Postgres code base.
+
+## What is Zig
+
+[Zig](https://ziglang.org/) is described on its website as a general-purpose programming language and toolchain for maintaining robust, optimal and reusable software. Zig is a new language (pre 1.0), but it has been [gaining popularity](https://notes.eatonphil.com/2024-03-15-zig-rust-and-other-languages.html) in the systems programming community. It is fair to think of it as a "modern C", providing safe(r) memory management, compilation time code execution (comptime), and a rich standard library. We'll show some of these features in the context of Postgres extensions in the next sections.
```

Comment:
safer is the key. Safer memory management and string handling. But Zig is no memory safe language like Go or Rust. 

Good example why Zig is not memory safe (author discovered what can go wrong when you move a self referential type to another memory location): https://zig.news/liyu1981/do-not-rely-on-the-pointer-to-content-of-returned-struct-1ijc

Reply:
Yeah, that's why I did the "safe(r)" thing. Maybe just "safer" would be more honest, though.